civis/polling.py

```diff
@@ -188,10 +195,14 @@ def cleanup(self):
                 self._polling_thread.cancel()
 
     def _reset_polling_thread(self,
-                              polling_interval=_DEFAULT_POLLING_INTERVAL):
+                              polling_interval=_DEFAULT_POLLING_INTERVAL,
+                              start_thread=False):
         with self._condition:
-            if self._polling_thread.is_alive():
+            if (getattr(self, '_polling_thread', None) is not None and
+                    self._polling_thread.is_alive()):
                 self._polling_thread.cancel()
             self.polling_interval = polling_interval
             self._polling_thread = _ResultPollingThread(self._check_result, (),
                                                         polling_interval)
+            if start_thread:
+                self._polling_thread.start()
```

Comment:
Could/should this line just be added wherever we need to start the thread rather than having an extra kwarg?  This way seems a little difficult to read, but it's very possible I'm missing something

Reply:
I added it here so that we centralize handling of the private attribute `_polling_thread`. There's only one place I'd need to add it (in the `ContainerFuture`) if it weren't here. Do you think that would be better?

Reply:
Hmm... I think either way is fine.